[DOCS] Endpoint artifact docs clarification (#2516)
elastic/security-docs: docs/management/admin/endpoint-artifacts.asciidoc

[[endpoint-artifacts]]
[chapter]
= Optimize {elastic-defend}

If you encounter problems like incompatibilities with other antivirus software, too many false positive alerts, or excessive storage or CPU usage, you can optimize {elastic-defend} to mitigate these issues.

Endpoint artifacts — such as trusted applications and event filters — and Endpoint exceptions let you modify the behavior and performance of _{elastic-endpoint}_, the component installed on each host that performs {elastic-defend}'s threat monitoring, prevention, and response actions.

The following table explains the differences between several Endpoint artifacts and exceptions, and how to use them:

[cols="2"]
|===

> | <<trusted-apps-ov,Trusted applications>>

Author: ```suggestion
| <<trusted-apps-ov,Trusted application>>
```

a| *_Prevents {elastic-endpoint} from monitoring a process._* Use to avoid conflicts with other software, usually other antivirus or endpoint security applications.

* Creates intentional blind spots in your security environment — use sparingly!

> * Does not monitor the application for threats, and does not create alerts for the application, even if it behaves like malware, ransomware, etc.

Reviewer: So "halts" makes it sound like it stops something that started. Slight revision suggestion: 
```suggestion
* Doesn't monitor the application for threats, nor does it generate alerts, even if it behaves like malware, ransomware, etc.
```

> * Does not generate events for the application except process events for visualizations.

Author: Someone on the dev side might answer this better, but I think a trusted app is about preventing event _generation_, not event _collection_. That's why it's such a blindspot: the event data just doesn't exist, because it's not getting created.

> * May improve performance, since {elastic-endpoint} monitors fewer processes.

Author: This revision makes it sound kind of intentional, like the trusted app reduces the number of processes _in order to_ improve performance. The performance boost is more of a side effect.

> * May still generate malicious behavior alerts, if the application's process events indicate malicious behavior.

Author: Yes, the minimal process events that still get created could possibly trigger the [malicious behavior protection](https://www.elastic.co/guide/en/security/master/configure-endpoint-integration-policy.html#behavior-protection) feature, probably because it's looking more at behavior patterns in the data, regardless of the application/process doing the behavior. Malicious behavior protection is separate from other Elastic Defend features that presumably would NOT trigger for a trusted application (such as malware, ransomware, or memory threat protections, etc.), since they wouldn't be looking at process events in the same way.

The "for visualizations" detail might be unnecessary and distracting here; we could probably just say "for internal use" or similar.

@caitlinbetz @kevinlog @ferullo Any additional details on what might trigger these malicious behavior alerts? I get the sense that we're mentioning this because it's happened to customers when they weren't expecting alerts, but any other info that would help?

> * May still generate malicious behavior alerts, if the application's process events indicate malicious behavior.

Author: @kevinlog Thanks for the extra context! I'll add a cross-reference to Endpoint alert exceptions as you suggest.

@nastasha-solomon I think this will resolve the point of confusion that customers may be having; it sounds like they might not need the additional explanation of how the alerts are still being created, just how they can stop them (create an exception).

> | <<event-filters,Event filters>>

Reviewer 2: ```suggestion
| <<event-filters,Event filter>>
```
The bullet points below this heading start with verbs that refer to singular nouns i.e. "Prevents". "Event filters... prevents" seems off to me. If you decide to include this change it applies to the "Trusted applications" and "Endpoint alert exceptions" sections too.

a| *_Prevents event documents from being written to {es}._* Use to reduce storage usage in {es}.

> Does NOT lower CPU usage for {elastic-endpoint}. It still monitors event data for possible threats, but the event data just isn't being written to {es}.

Reviewer 2: ```suggestion
Does NOT lower CPU usage for {elastic-endpoint}, which still monitors event data for possible threats, but without writing event data to {es}.
```

Author: I think the first statement has more punch as a separate sentence, but I think all the other stuff here.
```suggestion
Does NOT lower CPU usage for {elastic-endpoint}. It still monitors event data for possible threats, but without writing event data to {es}.
```

| <<blocklist,Blocklist>>
a| *_Prevents known malware from running._* Use to extend {elastic-defend}'s protection against malicious processes.

> NOT intended to generically block applications assumed to be benign.

Reviewer 2: Still confused by this. Why would someone think the blocklist was intended for blocking benign applications? I think maybe the word "generically" is doing some heavy lifting here that is going over my head?

Author: My response above: https://github.com/elastic/security-docs/pull/2516/files#r992703486

> | <<endpoint-rule-exceptions,Endpoint alert exceptions>>

Author: ```suggestion
| <<endpoint-rule-exceptions,Endpoint alert exception>>
```

> a| *_Prevents {elastic-endpoint} from generating alerts or stopping processes._* Use to reduce false positive alerts or preventions.

Reviewer 3: Ah ok, I didn't know we were referring to this feature as "preventions". Thanks for the clarification!
